src/steadystateproblem.cpp

```diff
@@ -209,6 +203,69 @@ void SteadystateProblem::findSteadyStateBySimulation(Solver *solver,
     }
 }
 
+bool SteadystateProblem::initializeBackwardProblem(Solver *solver,
+                                                   Model *model,
+                                                   const BackwardProblem *bwd) {
+    if (bwd) {
+        /* If preequilibration but not adjoint mode, there's nothing to do */
+        if (solver->getSensitivityMethodPreequilibration() !=
+            SensitivityMethod::adjoint)
+            return false;
+
+        /* If we need to reinitialize solver states, this won't work yet. */
+        if (model->nx_reinit() > 0)
+            throw NewtonFailure(AMICI_NOT_IMPLEMENTED,
+                "Adjoint preequilibration with reinitialization of "
+                "non-constant states is not yet implemented. Stopping.");
+
+        /* If we have a backward problem, we're in preequilibration.
+           Hence, quantities like t, x, and xB must be set. */
+        solver->reInit(t, x, x);
+        solver->updateAndReinitStatesAndSensitivities(model);
+        xB.copy(bwd->getAdjointState());
+    }
+
+    /* Will need to write quadratures: resize */
+    xQB.reset();
+    return true;
+}
+
+void SteadystateProblem::computeSteadyStateQuadrature(NewtonSolver *newtonSolver,
+                                                      Model *model) {
+    /* compute the integral over the adjoint state xB */
```

Comment:
huh, this code looks like it solves a linear system and does not compute an integral?

Reply:
I suppose this computes the quadratures but does not require integration in the steadystate case?

Reply:
This is what makes it nice:
In steady state, this integral has an analytical solution, which is given by the solution of this linear system, which is substantially faster than carrying out integration. However, if the Jacobian is singular, we have to carry out integration -> This is exactly what's planned for my next pr.

Reply:
Thats what I assumed, would be great if you could give a brief documentation that it can be and why it can be computed analytically ;)